src/test/java/com/google/step/datamanager/DealManagerDatastoreTest.java

```diff
@@ -0,0 +1,155 @@
+package com.google.step.datamanager;
+
+import static com.google.step.TestConstants.BLOBKEY_A;
+import static com.google.step.TestConstants.BLOBKEY_B;
+import static com.google.step.TestConstants.DATE_A;
+import static com.google.step.TestConstants.DATE_B;
+import static com.google.step.TestConstants.DATE_C;
+import static com.google.step.TestConstants.DATE_D;
+import static com.google.step.TestConstants.DEAL_ID_A;
+import static com.google.step.TestConstants.DESCRIPTION_A;
+import static com.google.step.TestConstants.DESCRIPTION_B;
+import static com.google.step.TestConstants.RESTAURANT_ID_A;
+import static com.google.step.TestConstants.RESTAURANT_ID_B;
+import static com.google.step.TestConstants.SOURCE_A;
+import static com.google.step.TestConstants.SOURCE_B;
+import static com.google.step.TestConstants.USER_ID_A;
+import static com.google.step.TestConstants.USER_ID_B;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.step.model.Deal;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class DealManagerDatastoreTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private final DealSearchManager searchManager = mock(DealSearchManager.class);
+
+  private final DealManagerDatastore dealManagerDatastore = new DealManagerDatastore(searchManager);
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testCreate_success() {
+    Deal deal =
+        dealManagerDatastore.createDeal(
+            DESCRIPTION_A, BLOBKEY_A, DATE_A, DATE_B, SOURCE_A, USER_ID_A, RESTAURANT_ID_A);
+    assertEquals(DESCRIPTION_A, deal.description);
+    assertEquals(BLOBKEY_A, deal.photoBlobkey);
+    assertEquals(DATE_A, deal.start);
+    assertEquals(DATE_B, deal.end);
+    assertEquals(SOURCE_A, deal.source);
+    assertEquals(USER_ID_A, deal.posterId);
+    assertEquals(RESTAURANT_ID_A, deal.restaurantId);
+  }
+
+  @Test
+  public void testRead_success() {
+    Deal createdDeal =
+        dealManagerDatastore.createDeal(
+            DESCRIPTION_A, BLOBKEY_A, DATE_A, DATE_B, SOURCE_A, USER_ID_A, RESTAURANT_ID_A);
+    Deal deal = dealManagerDatastore.readDeal(createdDeal.id);
+    assertEquals(DESCRIPTION_A, deal.description);
```

Comment:
just a suggestion, maybe you could use assertThat(deal, samePropertyValuesAs(createdDeal); instead of all the assertEquals. you can use import static org.hamcrest.Matchers.samePropertyValuesAs;

Reply:
I can't seem to get samePropertyValuesAs to return false even if the object is completely different. I'll leave it as many assertEquals for now.

Reply:
Oh right i realised that samePropertyValuesAs only works for classes with getters

Reply:
Another workaround to this may be overriding equals method in deal class